src/MSALWrapper.Test/AuthFlow/AuthFlowExecutorTest.cs

```diff
@@ -742,16 +744,62 @@ public async Task HasThreeAuthFlows_Returns_Early_With_TokenResultAndErrors()
             resultList[1].Should().BeEquivalentTo(authFlowResult2, this.ExcludeDurationTimeSpan);
         }
 
+        [Test]
+        public async Task HasMultipleAuthFlows_Returns_Early_With_TimeoutException()
+        {
+            var timeoutError = new[]
+            {
+                new TimeoutException("Timeout exception"),
+            };
+
+            var errors1 = new[]
+            {
+                new Exception("Exception 1"),
+            };
+
+            var authFlowResult1 = new AuthFlowResult(null, errors1, "authFlow1");
+            var authFlowResult2 = new AuthFlowResult(null, timeoutError, "authFlow2");

```

Comment:
You induce a timeout error by creating and injecting a result with a `TimeoutException` in it, but isn't it the `TimeoutManager` that causes that to happen? I might have misread this, but I think you're circumventing the very thing you intend to test here. Could you achieve a similar test by creating a specialized `ITimeoutManager` class? 🤔